### PR TITLE
feat: always add twitter:card

### LINKED ISSUE #46: Add twitter:card meta property by default
This will ensure pages rendered by Helix will always look nice on Twitter.

### PR DESCRIPTION
**Pull request by rofe**
Fix #46

### CODE REVIEW

File: src/steps/extract-metadata.js [hunk 1]
```diff
@@ -184,10 +184,22 @@ export default function extractMetaData(state, req) {
...
+        }
+        return true;
+      });
   }
 
+  // default value for twitter:card (mandatory for rendering URLs as cards in tweets)
+  if (!meta['twitter:card']) {
+    meta['twitter:card'] = 'summary_large_image';
+  }
   if (meta.keywords) {
```
Comment (tripodsan) [inline src/steps/extract-metadata.js:203]: looks a bit complicated....

why not: add `twitter:card` to the `ALLOWED_RESPONSE_HEADERS`
and just do `meta['twitter:card'] = metaConfig['twitter:card'] || 'summary_large_image'; `

**Review by tripodsan — CHANGES_REQUESTED** [review]
simplify

File: src/steps/extract-metadata.js (hunk 1 above)
Comment (tripodsan) [reply src/steps/extract-metadata.js:203]: or better: 

define:

```
const IGNORED_CUSTOM_META = [...ALLOWED_RESPONSE_HEADERS, 'twitter:card'];

and use this in the filter.
```

Comment (rofe) [reply src/steps/extract-metadata.js:203]: > looks a bit complicated....

I thought so, too 😄

Comment (rofe) [reply src/steps/extract-metadata.js:203]: > ```js
> const IGNORED_CUSTOM_META = [...ALLOWED_RESPONSE_HEADERS, 'twitter:card'];
> ```

Ok, but we still want `twitter:card` to be overridable in metadata.xslx or metadata blocks. If we filter it out of the custom metadata completely without capturing the value, wouldn't we end up always using the default value?

Comment (tripodsan) [reply src/steps/extract-metadata.js:203]: > If we filter it out of the custom metadata completely without capturing the value, wouldn't we end up always using the default value?

no, that's only the list of properties to be ignored for the `meta.custom` (which you also filter currently)

Comment (tripodsan) [reply src/steps/extract-metadata.js:203]: like this:
```js
diff --git a/src/steps/extract-metadata.js b/src/steps/extract-metadata.js
index dd69985..bc12fbd 100644
--- a/src/steps/extract-metadata.js
+++ b/src/steps/extract-metadata.js
@@ -19,6 +19,8 @@ import {
 import { filterGlobalMetadata, toMetaName, ALLOWED_RESPONSE_HEADERS } from '../utils/metadata.js';
 import { childNodes } from '../utils/hast-utils.js';
 
+const IGNORED_CUSTOM_META = [...ALLOWED_RESPONSE_HEADERS, 'twitter:card'];
+
 /**
  * Cleans up comma-separated string lists and returns an array.
  * @param {string} list A comma-separated list
@@ -180,7 +182,7 @@ export default function extractMetaData(state, req) {
   if (Object.keys(metaConfig).length > 0) {
     // add rest to meta.custom
     meta.custom = Object.entries(metaConfig)
-      .filter(([name]) => !ALLOWED_RESPONSE_HEADERS.includes(name))
+      .filter(([name]) => !IGNORED_CUSTOM_META.includes(name))
       .map(([name, value]) => ({
         name,
         value,
@@ -188,6 +190,9 @@ export default function extractMetaData(state, req) {
       }));
   }
 
+  // default value for twitter:card (mandatory for rendering URLs as cards in tweets)
+  meta['twitter:card'] = metaConfig['twitter:card'] || 'summary_large_image';
+
   if (meta.keywords) {
     meta.keywords = toList(meta.keywords).join(', ');
   }
```

Comment (rofe) [reply src/steps/extract-metadata.js:203]: But then if I filter it out before, I won't be able to capture its value (if present)... I must be missing something 😆

Comment (rofe) [reply src/steps/extract-metadata.js:203]: The above results in a failed test when the value is supposed to be overridden:
```
Rendering
       Metadata
         renders meta tags from metadata html block:

      AssertionError [ERR_ASSERTION]: HTML differs.
      + expected - actual

       ...
       ty="og:image:secure_url"><meta content="
      -summary_large_image
      +summary
       " name="twitter:card"><meta content="Pri
```

Comment (rofe) [reply src/steps/extract-metadata.js:203]: Ah, forget what I said 🤦  I missed the fact that you are suggesting to read the override value from `metaConfig` instead!

File: src/steps/extract-metadata.js [hunk 2]
```diff
@@ -180,14 +182,25 @@ export default function extractMetaData(state, req) {
...
+        property: name.includes(':') && !name.startsWith('twitter:'),
+      }))
+      .filter(({ name, value }) => {
+        // move special case twitter:card to top level
+        if (name === 'twitter:card') {
+          meta[name] = value;
+          return false;
+        }
+        return true;
+      });
```
Comment (tripodsan) [inline src/steps/extract-metadata.js:198]: you don't need this

File: src/steps/extract-metadata.js [hunk 3]
```diff
@@ -180,14 +182,17 @@ export default function extractMetaData(state, req) {
   if (Object.keys(metaConfig).length > 0) {
     // add rest to meta.custom
     meta.custom = Object.entries(metaConfig)
-      .filter(([name]) => !ALLOWED_RESPONSE_HEADERS.includes(name))
+      .filter(([name]) => !IGNORED_CUSTOM_META.includes(name))
       .map(([name, value]) => ({
         name,
         value,
-        property: name.includes(':'),
+        property: name.includes(':') && !name.startsWith('twitter:'),
```
Comment (tripodsan) [inline src/steps/extract-metadata.js:189]: you don't need this anymore neither :-)

```suggestion
        property: name.includes(':'),
```

**Review by tripodsan — APPROVED** [review]

**Pull request merged**